etl/steps/data/garden/growth/2024-01-04/gdp_historical.py

```diff
@@ -0,0 +1,189 @@
+    tb_md = tb_md.drop(columns=["population"])
+
+    #############################
+
+    # Merge both MPD and WDI world estimations in different columns and add suffixes. This will be useful for the next step.
+    tb = tb_mpd.merge(tb_wdi, on="year", how="left", suffixes=("_mpd", "_wdi"), short_name="gdp_historical")
+
+    # Apply Maddison Project Database growth retroactively to YEAR_WDI_MPD WDI data
+    tb = create_estimations_from_growth(
+        tb=tb, var_list=VAR_LIST, reference_year=YEAR_WDI_MPD, reference_var_suffix="_mpd", to_adjust_var_suffix="_wdi"
+    )
+
+    # Concatenate this with original WDI data (data after YEAR_WDI_MPD)
+    tb = pr.concat([tb, tb_wdi[tb_wdi["year"] > YEAR_WDI_MPD]], ignore_index=True)
+
+    # Merge datasets to include Maddison Database, which will be used as reference for the next step
+    tb = tb.merge(tb_md, on="year", how="outer", suffixes=("", "_md"), sort=True)
+
+    # Apply Maddison Database growth retroactively to YEAR_MPD_MD estimations
+    tb = create_estimations_from_growth(
+        tb=tb, var_list=VAR_LIST, reference_year=YEAR_MPD_MD, reference_var_suffix="_md", to_adjust_var_suffix=""
+    )
+
+    # Round variables to address uncertainty on old estimations (previous to 1990)
+    tb["gdp"] = tb["gdp"].round(-ACCURACY_GDP).where(tb["year"] < YEAR_WDI_MPD, tb["gdp"])
+    tb["gdp_per_capita"] = (
+        tb["gdp_per_capita"].round(-ACCURACY_GDP_PER_CAPITA).where(tb["year"] < YEAR_WDI_MPD, tb["gdp_per_capita"])
+    )
+
+    return tb
+
+
+def create_estimations_from_growth(
+    tb: Table, var_list: list, reference_year: int, reference_var_suffix: str, to_adjust_var_suffix: str
+) -> Table:
+    """
+    Adjust estimations of variables according to the growth of a reference variable.
+
+    Parameters
+    ----------
+    tb : Table
+        Table that contains both the reference variable (the one the growth is extracted from) and the variable to be adjusted (the one the growth is applied to).
+    var_list : list
+        List of the variable types to be adjusted. In this project, ["gdp", "gdp_per_capita"]
+    reference_year : int
+        Reference year from which the growth will be applied retroactively.
+    reference_var_suffix : str
+        Suffix of the reference variable (the one the growth is extracted from). In this project, "_mpd" or "_md".
+    to_adjust_var_suffix : str
+        Suffix of the variable to be adjusted (the one the growth is applied to). In this project, "_wdi" or "".
+
+    Returns
+    -------
+    tb : Table
+        Table with the adjusted variables.
+    """
+    for var in var_list:
+        # Get value from the reference variable in the reference year
+        reference_value = tb.loc[tb["year"] == reference_year, f"{var}{reference_var_suffix}"].iloc[0]
+
+        # The scalar is the previous value divided by the reference variable. This is the growth that will be applied retroactively to the variable to be adjusted.
```

Comment:
I think it's the other way around -  reference variable divided by the previous value but I now understand what this function does a lot better. thanks for making these changes!

Reply:
Ah, it's only that way because I am going backwards. But for clarity, I have inverted and then multiplied by the value to adjust, which is the same thing